persistence-modules/spring-data-jpa-repo-3/src/main/java/com/baeldung/spring/data/jpa/naturalid/HotelRoomsService.java

```diff
@@ -0,0 +1,35 @@
+package com.baeldung.spring.data.jpa.naturalid;
+
+import com.baeldung.spring.data.jpa.naturalid.entity.ConferenceRoom;
+import com.baeldung.spring.data.jpa.naturalid.entity.GuestRoom;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.EntityManagerFactory;
+import org.hibernate.Session;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+@Service
+public class HotelRoomsService {
+
+	private final EntityManager entityManager;
+
+	public HotelRoomsService(EntityManagerFactory entityManagerFactory) {
+		this.entityManager = entityManagerFactory.createEntityManager();
+	}
+
+	public Optional<ConferenceRoom> conferenceRoom(String name) {
+		Session session = entityManager.unwrap(Session.class);
+		return session.bySimpleNaturalId(ConferenceRoom.class)
+		  .loadOptional(name);
+	}
+
+	public Optional<GuestRoom> guestRoom(int roomNumber, int floor) {
+		Session session = entityManager.unwrap(Session.class);
+		return session.byNaturalId(GuestRoom.class)
+		  .using("roomNumber", roomNumber)
+		  .using("floor", floor)
+		  .loadOptional();
+	}
```

Comment:
Need to use spaces to indent instead of tabs in all the Java files.

Reply:
hello Kevin, I have just realized that I have used the wrong IntellJ formater, so I've re-formatted everything now.

In regards to this line in particular, I had a comment for one of my previous reviews stating that I should always use 2 spaces for the line continuations (both for git and code snippets in the article), even if this would mean manually adding the 2 spaces. How should I keep it?

Reply:
Either way is fine in the case of chaining method calls, which is different from a continuation. Some editors prefer two spaces in this case, while the guideline actually calls for four. An example of a continuation is a statement with a lot of operators that you spread over multiple lines (think concatenating a bunch of strings). Another is a method call where you want to put the parameters on separate lines.